### ISSUE TITLE
no-obj-calls does not report errors for `Reflect`

### ISSUE THREAD

**Issue by not-an-aardvark** [eslint/eslint#7700]
**Tell us about your environment**

* **ESLint Version:** 3.11.1
* **Node Version:** 7.2.0
* **npm Version:** 3.10.9

**What parser (default, Babel-ESLint, etc.) are you using?**

default

**Please show your full configuration:**

```yml
rules:
  no-obj-calls: error
```

**What did you do? Please include the actual source code causing the issue.**

```js
Reflect();
```

**What did you expect to happen?**

I expected ESLint to report an error about the `Reflect()` call, because it's a global object property and not a function.

**What actually happened? Please include the actual, raw output from ESLint.**

ESLint did not report any errors. Presumably, this is because the [`no-obj-calls`](https://github.com/eslint/eslint/blob/master/lib/rules/no-obj-calls.js) hasn't been updated to account for the new globals that were introduced in ES6.
